modules/opentelemetry/src/main/scala/zio/opentelemetry/CurrentSpan.scala

```diff
@@ -0,0 +1,91 @@
+package zio.opentelemetry
+
+import io.opentelemetry.common.AttributeValue
+import io.opentelemetry.context.propagation.HttpTextFormat
+import io.opentelemetry.trace.Span
+import zio.clock.Clock
+import zio.opentelemetry.ContextPropagation.{ extractSpan, injectSpan }
+import zio.opentelemetry.OpenTelemetry.{ createChildOf, currentSpan }
+import zio.opentelemetry.attributevalue.AttributeValueConverter
+import zio.opentelemetry.attributevalue.AttributeValueConverter.toAttributeValue
+import zio.{ RIO, URIO, ZIO }
+
+import scala.jdk.CollectionConverters._
+
+object CurrentSpan {
+
+  /**
+   * Gets the current span.
+   */
+  def getCurrentSpan: ZIO[OpenTelemetry, Nothing, Span] = currentSpan.flatMap(_.get)
+
+  /**
+   * Sets the current span.
+   */
+  def setCurrentSpan(span: Span): ZIO[OpenTelemetry, Nothing, Unit] = currentSpan.flatMap(_.set(span))
+
+  /**
+   * Extracts the span from carrier `C` and set its child span with name 'spanName' as the current span.
+   */
+  def createChildFromExtracted[C](
```

Comment:
Upon reading the documentation more, I realized that we probably should not expose any functionality to the end-user, that allows setting the span of the `FiberRef` without reverting it to the 'old' span. That probably this function, together with all other functions that set the span in the `FiberRef` (except those in `SpanSyntax`) should be hidden from the user?